test/uk/gov/hmrc/cdsreimbursementclaimfrontend/controllers/rejectedgoodssingle/EnterClaimControllerSpec.scala

```diff
@@ -412,6 +412,36 @@ class EnterClaimControllerSpec
           }
       }
 
+      "accept claim amount, for the full amount, when only a single claim is present and move on to total reimbursement page" in forAll {
+        (ndrcDetails: NdrcDetails, displayDeclaration: DisplayDeclaration) =>
+          whenever(ndrcDetails.amount.toInt > 11) {
```

Comment:
Is it safe to use? According with docs it throws DiscardedEvaluationException.
Proper way is whether to use `suchThat` for ScalaCheck generated values or just simply set required fields for generated values, for example, amount using copy - that one will be even faster as will help to avoid number of the missed.

Reply:
The recommended way to do property based testing with ScalaTest is to use `whenever` https://www.scalatest.org/user_guide/generator_driven_property_checks

Reply:
This is the link on ScalaTest wrapper over ScalaCheck and as a wrapper over some library the best thing they can do is to let generators to generate improper value and then throw exception (of course they will demonstrate theirs APIs). 
With `suchThat` you can have an instant effect on generator asking to regenerate another value until it matches boundaries - so no exception etc.
But it is also not very perfect way as it might slow down tests, i.e., generator might generate a lot of invalid values till succeeds. Because of that I was also saying that perhaps just generating some value and then setting field, such as,`amount` manually will be better, as will make test succeed on the first go.
That was my point.

Reply:
So it is up to you and rest. My only concern is that over-use of whenever and suchThat might lead to random test run time increase.

Reply:
Firstly, ScalaCheck is used to generate the random values, ScalaTest is used to perform the property based testing. Neither one is a wrapper of the other. When you use `suchThat` you create new generators, when you use `whenever` you are checking the generated values to make sure they will work for your test. They perform different purposes.
What you are suggesting is that we create new generators for each test, I disagree with this. The generators that we have got are the only ones we need. In this case we only need to make sure that the value generated allows our test to work properly. We do not need a new generator just for this test.

Reply:
You can use ScalaCheck without ScalaTest easily.
 Just import:
 `org.scalacheck.Prop.forAll`.
And ScalaTest decorates it, go and check source code yourself.

I don't understand what are you talking about. 
Here is source code of whenever, this is exactly do what I was saying above and from functionality stand point it is the same as `suchThat`.
```
  implicit def assertingNatureOfFutureAssertion: WheneverAsserting[Future[Assertion]] { type Result = Future[Assertion] } = {
    new WheneverAsserting[Future[Assertion]] {
      type Result = Future[Assertion]
      def whenever(condition: Boolean)(fun: => Future[Assertion]): Future[Assertion] =
        if (!condition)
          throw new DiscardedEvaluationException
        else
          fun
    }
  }
```

What? New generators? Where did you pick this one.
All I was saying use `copy` to set field like that:
```
forAll(someGenerator) { x =>
   x.copy(amount = 11)
}
```
Hope it is clear now.